apps/studio/src/components/ResourceSelector/ResourceSelector.tsx

```diff
@@ -1,157 +1,220 @@
-import { Suspense, useState } from "react"
-import {
-  Box,
-  Flex,
-  HStack,
-  Icon,
-  Skeleton,
-  Spacer,
-  Text,
-} from "@chakra-ui/react"
+import { Suspense, useEffect, useState } from "react"
+import { Box, Flex, HStack, Skeleton, Spacer, Text } from "@chakra-ui/react"
 import { Button, Link } from "@opengovsg/design-system-react"
-import { ResourceType } from "~prisma/generated/generatedEnums"
+import { ResourceType } from "@prisma/client"
 import { BiHomeAlt, BiLeftArrowAlt } from "react-icons/bi"
 
+import type { PendingMoveResource } from "~/features/editing-experience/types"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { sitePageSchema } from "~/pages/sites/[siteId]"
 import { trpc } from "~/utils/trpc"
 import { ResourceItem } from "./ResourceItem"
 
+interface ResourceSelectorProps {
+  queryFn:
+    | typeof trpc.resource.getChildrenOf.useInfiniteQuery
+    | typeof trpc.resource.getFolderChildrenOf.useInfiniteQuery
+  onChange: (resourceId: string) => void
+  selectedResourceId?: string
+  existingResource?: PendingMoveResource | null
+}
+
+const generatePermalinkPrefix = (parents: PendingMoveResource[]) => {
+  return parents.map((parent) => parent.permalink).join("/")
+}
+
 const SuspensableResourceSelector = ({
-  siteId,
-  selectedResourceId,
+  queryFn,
   onChange,
-  isDisabledFn,
+  selectedResourceId,
+  existingResource,
 }: ResourceSelectorProps) => {
-  const [ancestryStack] = trpc.resource.getAncestryOf.useSuspenseQuery({
-    siteId,
-    resourceId: selectedResourceId,
-  })
-  const [parentIdStack, setParentIdStack] = useState<string[]>(
-    ancestryStack.map((item) => item.id),
+  // NOTE: This is the stack of user's navigation through the resource tree
+  // NOTE: We should always start the stack from `/` (root)
+  // so that the user will see a full overview of their site structure
+  const [resourceStack, setResourceStack] = useState<PendingMoveResource[]>([])
+  const [isResourceHighlighted, setIsResourceHighlighted] =
+    useState<boolean>(true)
+  const { siteId } = useQueryParse(sitePageSchema)
+
+  const moveDest = resourceStack[resourceStack.length - 1]
+  const parentDest = resourceStack[resourceStack.length - 2]
+  const curResourceId = moveDest?.resourceId
```

Comment:
no action needed - some repeated code here which we could separate into a custom hook (`useMoveState` for example) but no urgency because

1. this likely hasn't been finalised yet, so behaviour can still differ
2. only 2 call sites (here + move modal) so the need for a custom hook is still low.

Reply:
@seaerchin have refactored such that this is also being used by "move modal" so it's only happening once